addons/resource/models/resource.py

```diff
@@ -944,16 +944,16 @@ def _adjust_to_calendar(self, start, end):
         end, revert_end_tz = make_aware(end)
         result = {}
         for resource in self:
-            calendar_start = resource.calendar_id._get_closest_work_time(start, resource=resource)
```

Comment:
_get_closest_work_time will convert start into the resource timezone to search for the work interval starting at the closest time of the datetime.
For a resource located in Asia/Calcutta, it will be:
start: day-1, 22:00 + 5:30 => day, 3:30
The closest start of the shift found will be the correct morning shift on the selected day (start = day, 8:30 in resource timezone)